pkg/dns/config/config.go

```diff
@@ -130,3 +132,39 @@ func (config *Config) validateUpstreamNameserver() error {
 	}
 	return nil
 }
+
+// ValidateNodeLocalCacheConfig returns nil if the config can be compiled
+// to a valid Corefile.
+func (config *Config) ValidateNodeLocalCacheConfig() error {
+	for _, stubDomains := range config.StubDomains {
+		if err := validateForwardProxy(stubDomains...); err != nil {
+			return err
+		}
+	}
+	if err := validateForwardProxy(config.UpstreamNameservers...); err != nil {
+		return err
+	}
+	return nil
+}
+
+// validateForwardProxy returns nil if the nameservers are valid proxy addresses
+// for the CoreDNS plugin forward.
+// The function is ported from coredns/plugin/forward:parseStanza
+func validateForwardProxy(nameservers ...string) error {
+	if len(nameservers) == 0 {
+		return nil
+	}
+	hosts, err := parse.HostPortOrFile(nameservers...)
+	if err != nil {
+		return err
+	}
+	for _, host := range hosts {
+		trans, _ := parse.Transport(host)
+		switch trans {
+		case "dns", "tls":
```

Comment:
As the coredns version used is not the latest AFAIK https://github.com/kubernetes/dns/blob/1fb1456f536245a6fd09622b84e6a7e918287f00/go.mod#L7

Could you verify that "dns"/"tls" protocols are supported in that older coredns version. So that we can allow it here in out validation?

Reply:
Yes, it's also valid in the older version: https://github.com/coredns/coredns/blob/v1.7.0/plugin/forward/setup.go#L101